tofu/geom/_GG.pyx

```diff
@@ -925,218 +883,366 @@ def discretize_vpoly(double[:,::1] VPoly, double dL,
-    Z, dZr, indZ, NZ = discretize_line1d(ZMinMax, dZ, DZ, Lim=True,
-                                          margin=margin)
-    Rn = len(R)
-    Zn = len(Z)
-
-    # Get the limits if any (and make sure to replace them in the proper
-    # quadrants)
+    # .. First we discretize R without limits ..................................
+    _st.cythonize_subdomain_dl(None, limits_dl) # no limits
+    _ = _st.discretize_line1d_core(&RMinMax[0], rstep, limits_dl,
+                                   True, 0, # discretize in absolute mode
+                                   margin, &disc_r0, reso_r0, &lindex,
+                                   ncells_r0)
+    free(lindex) # getting rid of things we dont need
+    lindex = NULL
+    # .. Now the actual R limited  .............................................
+    _st.cythonize_subdomain_dl(DR, limits_dl)
+    sz_r = _st.discretize_line1d_core(&RMinMax[0], rstep, limits_dl,
+                                      True, 0, # discretize in absolute mode
+                                      margin, &disc_r, reso_r, &lindex,
+                                      ncells_r)
+    free(lindex) # getting rid of things we dont need
+    # .. Now Z .................................................................
+    _st.cythonize_subdomain_dl(DZ, limits_dl)
+    sz_z = _st.discretize_line1d_core(&ZMinMax[0], zstep, limits_dl,
+                                      True, 0, # discretize in absolute mode
+                                      margin, &disc_z, reso_z, &lindex_z,
+                                      ncells_z)
+    # .. Preparing for phi: get the limits if any and make sure to replace them
+    # .. in the proper quadrants ...............................................
     if DPhi is None:
-        DPhi0, DPhi1 = -Cpi, Cpi
+        min_phi = -Cpi
+        max_phi = Cpi
     else:
-        DPhi0 = Catan2(Csin(DPhi[0]), Ccos(DPhi[0]))
-        DPhi1 = Catan2(Csin(DPhi[1]), Ccos(DPhi[1]))
-
-    dRPhir, dPhir = np.empty((Rn,)), np.empty((Rn,))
-    Phin = np.empty((Rn,),dtype=int)
-    NRPhi = np.empty((Rn,))
-    NRPhi0 = np.zeros((Rn,),dtype=int)
-    nRPhi0, indR0ii = 0, 0
-    NP, NPhimax = 0, 0
-    Rratio = int(Cceil(R[Rn-1]/R[0]))
-    for ii in range(0,Rn):
+        min_phi = DPhi[0] # to avoid conversions
+        min_phi = Catan2(Csin(min_phi), Ccos(min_phi))
+        max_phi = DPhi[1] # to avoid conversions
+        max_phi = Catan2(Csin(max_phi), Ccos(max_phi))
+    # .. Initialization ........................................................
+    sz_phi = <long*>malloc(sz_r*sizeof(long))
+    tot_nc_plane = <long*>malloc(sz_r*sizeof(long))
+    ncells_rphi  = <long*>malloc(sz_r*sizeof(long))
+    step_rphi    = <double*>malloc(sz_r*sizeof(double))
+    reso_phi = np.empty((sz_r,)) # we create the numpy array
+    reso_phi_mv = reso_phi # and its associated memoryview
+    r_ratio = <int>(Cceil(disc_r[sz_r - 1] / disc_r[0]))
+    twopi_over_dphi = _TWOPI / phistep
```

Comment:
I see you have changed the definition of this variable, phistep really is an agnle (and not a distance like before).
=> How does it affect the discretization ?
The idea was to have a phi discretization that would preserve a distance whatever the R. Is this still the case ?

If so, you have to convert it to a distance at some point, which value of R do you choose for that ?
If possible it would be nice if the user could choose directly the resolution in terms of distance (r*phi).
Otherwise, the user does not know which distance his choice of phistep corresponds to...